database/engine/rrdengineapi.c

```diff
@@ -195,20 +216,44 @@ void rrdeng_store_metric_flush_current_page(RRDDIM *rd)
     handle->descr = NULL;
 }
 
-void rrdeng_store_metric_next(RRDDIM *rd, usec_t point_in_time, NETDATA_DOUBLE n, SN_FLAGS flags)
+void rrdeng_store_metric_next(STORAGE_COLLECT_HANDLE *collection_handle, usec_t point_in_time, NETDATA_DOUBLE n,
+                              NETDATA_DOUBLE min_value,
+                              NETDATA_DOUBLE max_value,
+                              uint16_t count,
+                              uint16_t anomaly_count,
+                              SN_FLAGS flags)
 {
-    storage_number number = pack_storage_number(n, flags);
+    UNUSED(min_value);
+    UNUSED(max_value);
+    UNUSED(count);
 
-    struct rrdeng_collect_handle *handle = (struct rrdeng_collect_handle *)rd->state->handle;
-    struct rrdengine_instance *ctx;
-    struct page_cache *pg_cache;
-    struct rrdeng_page_descr *descr;
-    storage_number *page;
-    uint8_t must_flush_unaligned_page = 0, perfect_page_alignment = 0;
+    struct rrdeng_collect_handle *handle = (struct rrdeng_collect_handle *)collection_handle;
+    struct rrdeng_metric_handle *metric_handle = (struct rrdeng_metric_handle *)handle->metric_handle;
+    struct rrdengine_instance *ctx = handle->ctx;
+    struct page_cache *pg_cache = &ctx->pg_cache;
+    struct rrdeng_page_descr *descr = handle->descr;
+    RRDDIM *rd = metric_handle->rd;
+    int tier = ctx->tier;
 
-    ctx = handle->ctx;
-    pg_cache = &ctx->pg_cache;
-    descr = handle->descr;
+    storage_number number;
+    storage_number_tier1_t number_tier1;
+
+    number = pack_storage_number(n, flags);
```

Comment:
Hi! Is this needed here, since it is done again below ?

Reply:
removed it